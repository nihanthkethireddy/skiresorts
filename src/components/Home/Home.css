.resorts {
  text-align: center;
}

.cardGrid {
  display: flex;
    justify-content: space-evenly;
    flex-wrap: wrap;
}