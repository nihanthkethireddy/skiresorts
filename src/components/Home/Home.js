import { useEffect, useState } from 'react';
import { SkiCard } from '../SkiCard';
import {SkiDetails} from '../SkiDetails';
import './Home.css';

function Home() {
  const [resorts, setResorts] = useState([1,2,3,4,5,6,7,8,9,10])
  const [popup, setPopup] = useState({ open: false, resort: {}, mode: null })

  const displayResorts = () => {
    if (resorts?.length > 0) {
      return resorts.map((resort) => {
        return <SkiCard resort={resort} setPopup={setPopup} deleteResort={deleteResort} />
      })
    }
    return <div>No Resorts Found</div>
  }

  const displayPopup = () => {
    if (popup.open) {
      return <SkiDetails resort={popup.resort} deleteResort={deleteResort} updateResort={updateResort} mode={popup.mode} setPopup={setPopup}/>
    }
    return null
  }

  const deleteResort = (resort) => {
    console.log('deleted')
  }

  const updateResort = (resort) => {
    console.log('updated')
  }

  return (
    <div className="resorts">
      {displayPopup()}
      <div>Ski Resorts <button onClick={() =>  setPopup({open: true, resort: {}, mode: 'edit'})}>Add Resort</button></div>
      <div className={'cardGrid'}>{displayResorts()}</div>
    </div>
  );
}

export {Home};
