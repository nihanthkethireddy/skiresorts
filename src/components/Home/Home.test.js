import { shallow } from 'enzyme';
import { Home } from './Home';

describe('<Home /> with no props', () => {
  const container = shallow(<Home />);
  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  });

  it('Check if Add Resort button is rendered', () => {
    expect(container.find('button').length).toEqual(1);
  });

});