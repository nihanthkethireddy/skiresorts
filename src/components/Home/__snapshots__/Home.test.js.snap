// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> with no props should match the snapshot 1`] = `"<div class=\\"resorts\\"><div>Ski Resorts <button>Add Resort</button></div><div class=\\"cardGrid\\"><div>No Resorts Found</div></div></div>"`;
